docs/features/communication/ipc/index.rst

```diff
@@ -11,6 +11,397 @@
    #
    # SPDX-License-Identifier: Apache-2.0
    # *******************************************************************************
+   #
+   # Authors
+   # @hartmannnico, 
+
+
+Communication Framework
+#######################
+
+The Communication Framework handles the safe, secure and performant exchange of information between software and/or hardware components of the Score stack.
+
+Communiction covers information exchange between endpoints residing in
+* the same or different process, running on
+* the same or different operating systems, deployed on
+* the same or different compute devices, built into
+* the same or different devices.
+
+In current communication designs (SomeIP, Protobuf, Zenoh) communication on networks is at the centerpoint of considerations. We believe this to be the wrong approach for software defined machines. Network based communication circles around protocols based on a wire based communication paradigm, requiring serialization and segmentation of data.
+
+Instead, we promote a memory centered core paradigm and put Inter-Process Communication (IPC) into our conceptional focus. This allows for true zero copy design, does not segment or serialize data and fosters significantly higher performance in algorithms. As a bonus, it provides easy gateway mechanisms into serial communication like Ethernet and CAN.
+
+The communication framework is independent of the domain of the application it is used in.
+
+General Architecture
+--------------------
+
+The communication framework defines two central groups of elements:
+
+* Information Elements
+* Infrastructure Elements
+
+While the former carry the information exchanged the latter provide the means through which the information is exchanged.
+
+Information Elements
+````````````````````
+
+There are three fundamental concepts of information exchange defined. One communication element represents each concept:
+
+* **Topics**: A topic is the information carrier for **data**. A unique Id identifies a topic while a *data type* defines it's memory layout. The topic carries zero or multiple *values*. A value represents a single instances of the data type. See _`Topics`, _`Names`, _`Data Types`.
```

Comment:
Is this something with is aligned to existent technologies like Iceoryx, uprotocol etc.? I would propose that it would make sense to check which technologies are already there and not to start from scratch. We can also discuss this in the first IPC incubation round.

Reply:
Yes and no. It is in line with current available communication systems like iceoryx2 (in fact, they did exactly this). Yet I disagree with the argumentation. This is a contribution request to ask for what we would like to have. In that sense, it does not matter how exactly it matches with existing technologies which have proven to be not sufficient.